### PR TITLE
MQTT fix

### PR DESCRIPTION
**Pull request by Trickx**
- Fixed crash if MQTT broker is not present.
- Added debuglevel check for printf in ade7878.go

### CODE REVIEW

File: src/main/readout.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 	"math"
 	"os"
 	"path/filepath"
-	"github.com/nDenerserve/SmartPi/src/smartpi"
```
Comment (SuperQ) [inline src/main/readout.go]: I have a fix for the makefile so you don't need to change this.  Should also help with the go get a bit.

File: src/main/readout.go [hunk 2]
```diff
@@ -140,17 +140,47 @@ func updateSQLiteDatabase(c *smartpi.Config, data []float32) {
...
+					} else if token.Error() != nil {
+						if c.DebugLevel > 0 {
+							fmt.Println(token.Error())
+						}
+						status = false
+					}
 				}
 			}
+			if c.DebugLevel > 0 {
+				fmt.Println("done.")
```
Comment (SuperQ) [inline src/main/readout.go]: It would be nice to have a little more context than "done" in the log line.

Comment (Trickx) [inline src/main/readout.go]: It completes the log messages for the MQTT publications.
"Done" indicates fully processed.
Be aware of the indenting.

File: src/main/readout.go (hunk 1 above)
Comment (Trickx) [inline src/main/readout.go]: For the moment this did not work for me.

Comment (SuperQ) [inline src/main/readout.go]: Yea, you need to remove the `GOPATH` stuff from the makefile.

Comment (Trickx) [inline src/main/readout.go]: Typically, I'm working on my own fork.
Including an absolute path to github breaks my system.
How to handle?

Comment (SuperQ) [inline src/main/readout.go]: https://github.com/SuperQ/SmartPi/commit/e1bca4e25a4cfe458efd9c0c4a09cba376735c66

The only annoying thing is you'll need to symlink your fork into `${GOPATH}/src/github.com/nDenerserve/`. for all the code to work on your branch.

Comment (Trickx) [inline src/main/readout.go]: Annoying, yes and most of all unnecessary.
What is the benefit of this approach?
The old way I could just clone (my fork) and build via make.
Now, I have to fix all these unnecessary stuff without any benefit.

Comment (nD-enerserve) [inline src/main/readout.go]: Ok. With the makefile it works without github.com/nDenerserve/SmartPi/src/

**Pull request merged**